java-checks/src/main/java/org/sonar/java/checks/InterruptedExceptionCheck.java

```diff
@@ -131,16 +137,31 @@ public BlockVisitor(Symbol catchedException) {
 
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
-      Symbol symbol = tree.symbol();
-      if (!symbol.isUnknown() && symbol.owner().type().isSubtypeOf("java.lang.Thread") && "interrupt".equals(symbol.name()) && tree.arguments().isEmpty()) {
+      if (threadInterrupted || depth >= MAX_DEPTH) {
```

Comment:
Right now, when the maximum depth is reached, we stop looking at any method calls (including interrupt) and at thrown exceptions. This means that if we're at depth `MAX_DEPTH - 1` and see a call to another method, we will then visit that method, but won't look at anything inside that method. So visiting that method would be pointless.

What I think should be happening is that we stop recursing into new methods when the maximum depth is reached, but we still look for `interrupt` calls and `throw`s in the method that reached the final depth. For that purpose, this if statement and the following `depth++` should be moved to after the `INTERRUPT_MATCHERS` if-statement and the `depth` check in `visitThrowStatement` should be removed.